packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -1086,15 +1081,34 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                     if (handlePath.length > 0 && !handlePath.startsWith("/")) {
                         handlePath = `/${handlePath}`;
                     }
-                    id = `${parentHandle}/.app${handlePath}`;
-
-                    // TODO: SPO will deprecate this soon
-                    if (summaryObject.handleType === api.SummaryType.Commit) {
-                        value = {
-                            content: id,
-                        };
+                    const pathKey = `.app${handlePath}`;
+                    // We try to get the summary tree from the cache so that we can expand it in order to dedup the blobs.
+                    // We always send whole tree no matter what, even if some part of the tree did not change in order to dedup
+                    // the blobs. However it may happen that the tree is not found in cache and then we have to use the handle.
+                    const summaryTreeToExpand = this.treesPathToSummaryTreeCache.get(pathKey);
```

Comment:
I believe summaryTreeToExpand should never be undefined, if so, asserting it would be great!

Reply:
If not, when can it be undefined? Client should only reference previous summary paths, and they all should be known to us, right?

Reply:
We were clearing the caches in this condition
if (context.proposalHandle !== this.blobsShaProposalHandle) {
So in that case we would have reached to else part but now I have changed that logic and using a backup cache. So we should not reach here.
But before removing it, I would keep it and add telemetry so as to make sure we never reach here. Because this is new code and could affect documents I don't want to remove it  right away. Keeping it will let us know that our code is right without affecting the docs.

Reply:
Added to telemetry event just to for caution.